spec/models/text_annotation_spec.rb

```diff
@@ -4,8 +4,50 @@
 
     it { is_expected.to validate_presence_of(:line_start) }
     it { is_expected.to validate_presence_of(:line_end) }
+    it { is_expected.to validate_presence_of(:column_start) }
+    it { is_expected.to validate_presence_of(:column_end) }
+
+    it { is_expected.to allow_value(1).for(:line_start) }
+    it { is_expected.to allow_value(1).for(:line_end) }
+    it { is_expected.to allow_value(0).for(:column_start) }
+    it { is_expected.to allow_value(0).for(:column_end) }
+
     it { is_expected.to allow_value(10).for(:line_start) }
     it { is_expected.to allow_value(10).for(:line_end) }
+    it { is_expected.to allow_value(5).for(:column_start) }
+    it { is_expected.to allow_value(5).for(:column_end) }
+
+    it { is_expected.not_to allow_value(0).for(:line_start) }
+    it { is_expected.not_to allow_value(0).for(:line_end) }
+    it { is_expected.not_to allow_value(-1).for(:line_start) }
+    it { is_expected.not_to allow_value(-1).for(:line_end) }
+    it { is_expected.not_to allow_value(-1).for(:column_start) }
+    it { is_expected.not_to allow_value(-1).for(:column_end) }
+
+    it { is_expected.not_to allow_value(nil).for(:line_start) }
+    it { is_expected.not_to allow_value(nil).for(:line_end) }
+    it { is_expected.not_to allow_value(nil).for(:column_start) }
+    it { is_expected.not_to allow_value(nil).for(:column_end) }
+
+    it 'is invalid with line_start less than 1' do
```

Comment:
You don't need the tests here (extending to line 50), it's just a longer form of syntax for the `is_expected` tests above, which are covering all of the cases